eventlet/green/ssl.py

```diff
@@ -40,6 +39,63 @@ class GreenSSLSocket(_original_sslsocket):
     settimeout(), and to close/reopen the connection when a timeout
     occurs at an unexpected juncture in the code.
     """
+
```

Comment:
I respect your decision to use a context manager instead of `try` / `finally`. However, I have several things to say about `SwapBackOriginal`.

* Please don't nest it within `GreenSSLSocket`. Even though the distinction between `GreenSSLSocket` methods and `SwapBackOriginal` methods is clear to Python, it's not as clear to a human reader.
* I generally prefer to use a generator function decorated with [`contextlib.contextmanager`](https://docs.python.org/2/library/contextlib.html#contextlib.contextmanager) to express all but complex context managers -- I find that more readable.
* I would like your context manager even more if it were parameterized with the module name and (e.g.) a `**kwds` parameter with which to specify the symbols to save/restore and the values with which to (temporarily) replace them. One of the benefits of using a `contextlib.contextmanager` function is that, of course, function parameters are straightforward.

It's possible that a generalized "temporarily use these instead" context manager could be useful in more places within `eventlet`, though I'm content to leave it in this module for now.